src/Components/Address/NewAddressForm.tsx

```diff
@@ -0,0 +1,261 @@
+
+  const serializedState = JSON.stringify({ values, errors, touched })
+  // hack to inspect/debug intermediate form state before submission
+  useEffect(() => {
+    props.onChange && props.onChange({ values, errors, touched })
+
+    // really props.onChange should be wrapped in useCallback before being passed in - then we would have this guarantee
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [serializedState])
+  return (
+    <Form>
+      <AddressInputs
+        endContent={
+          <Button type="submit" variant="primaryBlack" width="50%">
+            Save and Continue
+          </Button>
+        }
+      />
+    </Form>
+  )
+}
+
+export interface AddressFormWrapperProps<T extends AddressFormValues>
+  extends Omit<FormikConfig<T>, "initialValues"> {
+  initialValues: Partial<T>
+}
+
+/**
+ * A Formik context that provides a default validation schema and initial values
+ * for an address form. Provide a custom type extending `AddressFormValues` to
+ * add additional fields, along with other Formik props. The children should
+ * formik-compatible contain inputs for any AddressFormValues and other fields.
+ */
+export const AddressFormWrapper = <
+  T extends AddressFormValues = AddressFormValues
+>({
+  initialValues = {} as Partial<T>,
+  children,
+  validate,
+  validationSchema,
+  ...props
+}: AddressFormWrapperProps<T>): JSX.Element => {
+  let validationRule: Pick<FormikConfig<T>, "validate" | "validationSchema">
+  if (validationSchema) {
+    validationRule = { validationSchema }
+  } else if (validate) {
+    validationRule = { validate }
+  } else {
+    validationRule = { validationSchema: DEFAULT_VALIDATION_SCHEMA }
+  }
+
+  return (
+    <Formik<T>
+      initialValues={{
+        ...(EMPTY_ADDRESS as T),
+        ...initialValues,
+      }}
+      {...validationRule}
+      {...props}
+    >
+      {children}
+    </Formik>
+  )
+}
+
+/**
+ * An input layout that can be used in the context of a form to collect an address. It provides a default
+ * layout for the address fields and can be used with the `AddressFormWrapper`
+ * to provide default values and validation - or you can add your own additional
+ * values (include types that extend <AddressFormValues>) and override validation.
+ * @example
+ * ```tsx
+ *  const BasicExample: React.FC<{}> = () => {
+ *   return (
+ *     <AddressFormWrapper<AddressFormValues>
+ *       initialValues={initialAddress}
+ *       onSubmit={values => {
+ *         alert(JSON.stringify(values))
+ *       }}
+ *     >
+ *      <Form>
+ *       <BasicAddressForm />
+ *      </Form>
+ *     </AddressFormWrapper>
+ *   )
+ *  }
+ * ```
+ */
+export const AddressInputs: FC<{ endContent?: JSX.Element }> = ({
+  endContent,
+}) => {
+  const {
+    values,
+    errors,
+    touched,
+    status,
+    handleChange,
+    handleBlur,
+  } = useFormikContext<AddressFormValues>()
+  return (
+    <GridColumns data-testid="address-inputs">
```

Comment:
#non-blocking, but following the principle of 1:1, its best to keep this stuff all consistent. If this component is called `AddressInputs`, it follows that the `data-testid` should also equal `AddressInputs`. Then when searching for one thing, you find the other, and so forth. `address-inputs` is 1:2.